### PR TITLE
CLN: DataFrame.__repr__

### PR DESCRIPTION
**Pull request by topper-123**
Minor clean-up.

### CODE REVIEW

**Comment by jreback** [comment]
thanks @topper-123

**Comment by jbrockmendel** [comment]
Could be causing a mypy failure on master?

File: pandas/core/frame.py [hunk 1]
```diff
@@ -988,15 +988,13 @@ def __repr__(self) -> str:
...
         if self._info_repr():
+            buf = StringIO("")
             self.info(buf=buf)
             return buf.getvalue()
 
         repr_params = fmt.get_dataframe_repr_params()
-        self.to_string(buf=buf, **repr_params)
-
-        return buf.getvalue()
+        return self.to_string(**repr_params)
```
Comment (simonjayhawkins) [inline pandas/core/frame.py:997]: pandas/core/frame.py:997: error: Incompatible return value type (got "Optional[str]", expected "str")

**Comment by simonjayhawkins** [comment]
> Could be causing a mypy failure on master?

will revert for now to get ci to green.

**Comment by simonjayhawkins** [comment]
@topper-123 resubmit this when you can.

**Pull request merged**